akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -106,6 +106,12 @@ private[remote] object AssociationState {
   private final case class UniqueRemoteAddressValue(
       uniqueRemoteAddress: Option[UniqueAddress],
       listeners: List[UniqueAddress => Unit])
+
+  sealed trait UniqueRemoteAddressState
+  final case class UidKnown(uniqueAddress: UniqueAddress) extends UniqueRemoteAddressState
+  case object UidUnknown extends UniqueRemoteAddressState
+  final case class UidQuarantined(uniqueAddress: UniqueAddress) extends UniqueRemoteAddressState
```

Comment:
I think they may all be case objects. Whenever we pattern match on them, we are using `(_)` for the case classes. We simply don't use the value at all.

Reply:
um, I guess I was thinking of other imaginary usages of these. I'll change to objects.